src/go/rpk/pkg/api/admin/api_cloud_storage_test.go

```diff
@@ -0,0 +1,205 @@
+// Copyright 2023 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package admin
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type (
+	testRequestHandler func(w http.ResponseWriter, r *http.Request)
+	testFn             func(t *testing.T) testRequestHandler
+	requestTestCase    struct {
+		name             string
+		testFn           testFn
+		shouldError      bool
+		expectedErrorMsg string
+		expectedResponse interface{}
+	}
+)
+
+func TestStartAutomatedRecovery(t *testing.T) {
+	successfullStartResponse := RecoveryStartResponse{
+		Code:    200,
+		Message: "Automated recovery started",
+	}
+
+	ctx := context.Background()
+	runTest := func(t *testing.T, test requestTestCase) {
```

Comment:
[question] why do you choose to create `runTest` instead of:

```
t.Run(test.name, func(t *testing.T) {
	// put the logic here
})
``` 
?

Reply:
It's a combination of two personal preferences

1. To avoid nesting the code. I think it's easier to read where a scope begins and ends when the amount of indentation and nesting is limited.
2. I think it's easier to refactor code that is scoped into functions and that it makes maintainability easier in the long run. 
3. I tend to default to putting things into functions when I am iterating, to make it a bit easier on myself to adjust the code when I inevitably find out my initial assumptions about how to solve the problem were wrong.
4. I think it's a lot easier to unit test many small functions than a few big ones, which also makes me default to putting code into new functions early and often (not applicable here, since it is unit test code :sweat_smile:)

I know it's more of a "functional programming"-mentality and it might go against the coding style in the project in general, so if you think it's disruptive and makes the code-base harder to maintain overall, I'm happy to refactor it :blush:

Reply:
I'd also err on moving this function to the test code above, from a perspective of it being much more common and more in line with other tests in rpk.